marlin/src/lib.rs

```diff
@@ -127,7 +127,7 @@ impl<F: PrimeField, PC: PolynomialCommitment<F>, D: Digest> Marlin<F, PC, D> {
     pub fn index<'a, C: ConstraintSynthesizer<F>>(
         srs: &UniversalSRS<F, PC>,
         c: &C,
-    ) -> Result<(IndexProverKey<'a, F, PC, C>, IndexVerifierKey<F, PC, C>), Error<PC::Error>> {
```

Comment:
We should revert the changes to this file and `marlin/src/snark.rs`.

The `index` method should return a tuple of the indexed `ProvingKey` and indexed `VerificationKey`. These are uniquely indexed for a given `(SRS, Circuit)` tuple.

Alternatively, we may consider constructing two methods, such as `index_prover` and `index_verifier` or equivalent naming convention, to ensure we continue to provide complete functionality.

Reply:
I reverted those changes for now; it seemed to me that the verification key was just being duplicated there, but I was worried the resulting API change might be undesirable - yeah, we can revisit this later on.